jl4-core/src/L4/Evaluate.hs

```diff
@@ -632,6 +655,43 @@ runLit :: Lit -> Eval Value
 runLit (NumericLit _ann num) = pure (ValNumber num)
 runLit (StringLit _ann str)  = pure (ValString str)
 
+runBuiltin :: Stack -> [Value] -> UnaryBuiltinFun -> Eval Value
```

Comment:
same problem as in EvaluateLazy. I don't think all this repetition is good. 

I'm not too fond of DRY but if it's that much code it's almost impossible to keep them in sync.